### PR TITLE
[invoices] Feature async

### PR DESCRIPTION
**Pull request by osallou**
* manage invoices in background (to avoid long calculations delay)
*factorize invoice computations, were in 2 places in xxxinvoice controllers and xxinvoice in model depending on where request was done (global or specific)
* fixes on checks (global were checking only for booking clients)
* incoherent code between different invoices implementations

### CODE REVIEW

File: Framework/Events.php [hunk 1]
```diff
@@ -24,10 +25,16 @@
 require_once 'Modules/quote/Model/Quote.php';
 require_once 'Modules/services/Model/SeService.php';
 
+require_once 'Modules/invoices/Model/GlobalInvoice.php';
+
+require_once 'Modules/booking/Model/BookingInvoice.php';
+require_once 'Modules/services/Model/ServicesInvoice.php';
+
```
Comment (gaetanHerve) [inline Framework/Events.php:32]: missing import: require_once 'Modules/booking/Model/BookinginvoiceTranslator.php';

Comment (osallou) [reply Framework/Events.php:32]: not needed here, certainly elsewhere, i gonna look

Comment (osallou) [reply Framework/Events.php:32]: ah confusing with translators.... ok

Comment (osallou) [reply Framework/Events.php:32]: in bookinginvoice.php

File: Modules/booking/Model/BookingInvoice.php [hunk 2]
```diff
@@ -281,11 +465,11 @@ protected function getUnitPackagePricesForEachResource($id_space, $resources, $L
         $modelPriceOwner = new BkOwnerPrice();
         foreach ($resources as $resource) {
             // get the packages prices
-            $packages = $modelPackage->getByResource($id_space, $resource["id"]);
+            $packages = $modelPackage->getByResource($id_space ,$resource["id"]);
 
             $pricesPackages = array();
             for ($i = 0; $i < count($packages); $i++) {
-                $price = $modelPriceOwner->getPackagePrice($id_space, $packages[$i]["id"], $resource["id"], $id_unit);
+                $price = $modelPriceOwner->getPackagePrice($id_space, $packages[$i]["id"], $resource["id"], $id_client);
```
Comment (osallou) [inline Modules/booking/Model/BookingInvoice.php:472]: To check this modification, why changing unit to client, is it an error?

**Review by osallou — COMMENTED** [review]
To check

File: Modules/booking/Model/BookingInvoice.php (hunk 2 above)
Comment (osallou) [reply Modules/booking/Model/BookingInvoice.php:472]: nope, there is no unit defined

**Comment by gaetanHerve** [comment]
closes #657

**Pull request merged**